### PR TITLE
prohibited data and mhtml protocols

### PR DESCRIPTION
**Pull request by adon-at-work**
- prohibited data and mhtml protocols
- cleaner and smaller uri blacklist filter 
- shrinking the size of dist/xss-filters.min.js

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

**Comment by adon-at-work** [comment]
@neraliu, mind having a quick look on this?

File: src/xss-filters.js [hunk 1]
```diff
@@ -24,102 +24,39 @@ exports._getPrivFilters = function () {
...
-
-    // Given a full URI, need to support "[" ( IPv6address ) "]" in URI as per RFC3986
-    // Reference: https://tools.ietf.org/html/rfc3986
-    var URL_IPV6 = /\/\/%5[Bb]([A-Fa-f0-9:]+)%5[Dd]/;
+    var URI_BLACKLIST_PROTOCOLS = ['javascript', 'data', 'vbscript'],
+        URI_PROTOCOL_COLON = /(?::|&#[xX]0*3[aA];?|&#0*58;?|&colon;)/,
+        URI_PROTOCOL_HTML_ENTITIES = /&(?:#([xX][0-9A-Fa-f]+|\d+);?|Tab;|NewLine;)/g,
+        URI_PROTOCOL_WHITESPACES = /(?:^[\u0001-\u0020]+|[\t\n\r]+)/g,
```
Comment (neraliu) [inline src/xss-filters.js]: is it include all whitespaces?
https://developer.mozilla.org/en/docs/Web/JavaScript/Guide/Regular_Expressions#special-white-space

Comment (adon-at-work) [inline src/xss-filters.js]: I believe we'd refer to the whitespace list of different browsers instead of the whitespaces for regular expressions

**Comment by adon-at-work** [comment]
fixed https://github.com/yahoo/xss-filters/issues/12

**Pull request merged**